neo4j/query/query.go

```diff
@@ -19,4 +19,10 @@ const (
 	MarkNodesToRemain            = "MATCH (parent:`_hierarchy_node_%s_%s`)<-[:hasParent*]-(child:`_hierarchy_node_%s_%s`) WHERE child.hasData=true set parent.remain=true set child.remain=true"
 	RemoveNodesNotMarkedToRemain = "MATCH (node:`_hierarchy_node_%s_%s`) WHERE NOT EXISTS(node.remain) DETACH DELETE node"
 	RemoveRemainMarker           = "MATCH (node:`_hierarchy_node_%s_%s`) REMOVE node.remain"
+
+	HierarchyExists     = "MATCH (i:`_hierarchy_node_%s_%s`) RETURN i LIMIT 1"
+	GetHierarchyRoot    = "MATCH (i:`_hierarchy_node_%s_%s`) WHERE NOT (i)-[:hasParent]->() RETURN i LIMIT 1" // TODO check if this LIMIT is valid
```

Comment:
Outstanding TODO: Have we checked this limit?

Reply:
this TODO has been copied over from the original implementation. i havent checked the limit, but im also not concerned about it and dont understnad why it was a TODO in the first place. happy to remove it :)